src/snabbdom.ts

```diff
@@ -160,20 +163,21 @@ export function init(modules: Array<Partial<Module>>, domApi?: DOMAPI) {
                         startIdx: number,
                         endIdx: number): void {
     for (; startIdx <= endIdx; ++startIdx) {
-      let i: any, listeners: number, rm: () => void, ch = vnodes[startIdx];
+      let listeners: number, rm: () => void, ch = vnodes[startIdx];
       if (ch != null) {
         if (isDef(ch.sel)) {
           invokeDestroyHook(ch);
           listeners = cbs.remove.length + 1;
-          rm = createRmCb(ch.elm as Node, listeners);
-          for (i = 0; i < cbs.remove.length; ++i) cbs.remove[i](ch, rm);
-          if (isDef(i = ch.data) && isDef(i = i.hook) && isDef(i = i.remove)) {
-            i(ch, rm);
+          rm = createRmCb(ch.elm!, listeners);
+          for (let i = 0; i < cbs.remove.length; ++i) cbs.remove[i](ch, rm);
+          const removeHook = ch?.data?.hook?.remove;
+          if (isDef(removeHook)) {
+            removeHook(ch, rm);
```

Comment:
The current variable reuse pattern, is it for performance or for less code or what, please?

In any case, this :+1:

Reply:
Mostly for less code. In particular with the deep access of properties that might be undefined along the way, I found it to looks nice back when I wrote Snabbdom.

But we agree that it confusing and somewhat off-putting. We could probably cut back on it even more.